src/QueryFilterHelpers.ts

```diff
@@ -207,38 +218,58 @@ export function interpretExpression(
   parameters: any,
   patient: R4.IPatient
 ): AnyFilter {
+  let returnFilter: AnyFilter = {
+    type: ''
+  };
+  let errorOcurred = false;
   switch (expression.type) {
     case 'Equal':
-      return interpretEqual(expression as ELMEqual, library);
+      returnFilter = interpretEqual(expression as ELMEqual, library);
+      break;
     case 'Equivalent':
-      return interpretEquivalent(expression as ELMEquivalent, library);
+      returnFilter = interpretEquivalent(expression as ELMEquivalent, library);
+      break;
     case 'And':
-      return interpretAnd(expression as ELMAnd, library, parameters, patient);
+      returnFilter = interpretAnd(expression as ELMAnd, library, parameters, patient);
+      break;
     case 'Or':
-      return interpretOr(expression as ELMOr, library, parameters, patient);
+      returnFilter = interpretOr(expression as ELMOr, library, parameters, patient);
+      break;
     case 'IncludedIn':
-      return interpretIncludedIn(expression as ELMIncludedIn, library, parameters);
+      returnFilter = interpretIncludedIn(expression as ELMIncludedIn, library, parameters);
+      break;
     case 'In':
-      return interpretIn(expression as ELMIn, library, parameters);
+      returnFilter = interpretIn(expression as ELMIn, library, parameters);
+      break;
     case 'Not':
-      return interpretNot(expression as ELMNot);
+      returnFilter = interpretNot(expression as ELMNot);
+      break;
     case 'GreaterOrEqual':
-      return interpretGreaterOrEqual(expression as ELMGreaterOrEqual, library, parameters, patient);
+      returnFilter = interpretGreaterOrEqual(expression as ELMGreaterOrEqual, library, parameters, patient);
+      break;
     default:
       console.error(`Don't know how to parse ${expression.type} expression.`);
       // Look for a property (source attribute) usage in the expression tree. This can denote an
       // attribute on a resource was checked but we don't know what it was checked for.
       const propUsage = findPropertyUsage(expression, expression.localId);
+
       if (propUsage) {
         return propUsage;
       }
```

Comment:
Open question: I see you've changed the above cases to set the value of a variable, rather than return it.

Do we want this behavior here to? Or is the default a special case here where we'll always want to return immediately rather than do the logic at the bottom of this function. @hossenlopp might know.

Reply:
I don't understand why the changes above were made. It makes sense to return immediately. My worry is that that malformed filter with no `type` could be returned if it isn't replaced with a valid filter.